.github/workflows/publish-sql-schema.yml

```diff
@@ -24,8 +30,40 @@ jobs:
       run: |
         VERSION=${GITHUB_REF/refs\/tags\//}
         echo "MINOR_VERSION=$(echo ${VERSION} | awk -F'.' '{if (NF != 3) {exit 1}; printf "%s.%s", $1, $2}')" >> $GITHUB_OUTPUT
+    - name: Check validity of migrations
+      uses: "./.github/actions/validate-migrations"
+    - name: Check that existing migrations have not changed
+      if: '!inputs.force'
+      run: |
+        # Overwite the migrations on the local ref so that we can use `git diff` to compare.
+        rm -f db/*.sql
+        trap "git reset --hard" EXIT
```

Comment:
Perhaps we should run the entire script under `set -eo pipefail` so that we don't have to write traps?

Reply:
`run` implicitly runs with `set -eo pipefail`--this trap is more making sure that no matter what happens, i.e. script exiting on another command, the repo is restored. It's additional protection against copying bad state to GCS.

Reply:
Moot point now, as the new approach doesn't contaminate the working tree.